src/Symfony/Component/Filesystem/Watcher/INotifyWatcher.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Filesystem\Watcher;
+
+/**
+ * @internal
+ */
+class INotifyWatcher implements WatcherInterface
+{
+    public function watch($path, callable $callback)
+    {
+        $inotifyInit = inotify_init();
+        $dir = false;
+
+        stream_set_blocking($inotifyInit, 0);
```

Comment:
this means the call to `inotify_read()` below is non-blocking so that the `while` will burn the CPU, isn't it?